.github/workflows/release.yml

```diff
@@ -0,0 +1,35 @@
+# Copyright © 2022 The OSRD Contributors
+#
+# SPDX-License-Identifier: LGPL-3.0-only
+
+name: Publish Docker image on release
+
+on:
+  release:
+    types: [published]
+
+jobs:
+  build-and-push-docker-image:
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v3
+
+      - name: Set up Docker Buildx
+        uses: docker/setup-buildx-action@v2
+
+      - name: Log in to GitHub Container Registry
+        uses: docker/login-action@v2
+        with:
+          registry: ghcr.io
+          username: '$'
+          password: ${{ secrets.GITHUB_TOKEN }}
+
+      - name: Build Docker image
+        run: |
+          docker build -t ghcr.io/${{ github.repository_owner }}/osrd-stable/osrd-images
```

Comment:
You need to tag with the release number here (eg. `ghcr.io/${{ github.repository_owner }}/osrd-stable/osrd-images:v0.3.2`)